source/adapters/hip/common.hpp

```diff
@@ -204,3 +204,38 @@ template <typename T> class ReleaseGuard {
   /// UR object.
   void dismiss() { Captive = nullptr; }
 };
+
+// Helper method to return a pointer's attributes, or std::nullopt in the case
+// that the pointer is unknown to the HIP subsystem.
+inline static std::optional<hipPointerAttribute_t>
+getPointerAttributes(const void *pMem) {
+  // do not throw if hipPointerGetAttributes returns hipErrorInvalidValue
+  hipPointerAttribute_t hipPointerAttributes;
+  hipError_t Ret = hipPointerGetAttributes(&hipPointerAttributes, pMem);
+  if (Ret == hipErrorInvalidValue) {
```

Comment:
Last one, I just caught that - `hipErrorInvalidValue` can also be returned very early, before even checking the pointer base type, if either the ptr or the attributes passed are nullptr. In this case I think it we need to change the if-statement to
`if (Ret == hipErrorInvalidValue && pMem)` ensuring that the we are not going to skip erroring out on actual null memory.

Reply:
Thanks! I've also updated the function's comments to try and clarify that a null pointer is an error, rather than returning `std::nullopt`